src/charm.py

```diff
@@ -10,6 +10,18 @@
 from time import sleep
 from typing import Optional
 
+try:
+    # Library depends on 'cryptography', a platform-specific dependency.
+    # Importing it will fail on wrong architecture.
+    from charms.mysql.v0.tls import MySQLTLS
```

Comment:
This is the first import that I found fails due to the architecture it gets built on. It fails because it relies, indirectly, on the `cryptography` package. As an alternative, we could try to import `cryptography` directly, although Ruff would complaint that the import would not be used anywhere.

Let me know what you think.

Reply:
> I would check is_wrong_architecture before any imports

You mean even before the Standard library imports? Yeah, we can move it further up.

> I don't think we should raise since that will obscure the blocked status

The `raise` clause is there for cases where the import fails for any other reason (i.e. forget to change its name after a refactor, deploying in the wrong architecture but not actually executing the charm code, etc).

For the case that concern us, the `WrongArchitectureWarningCharm` class already exits with code `0` (see [code](https://github.com/canonical/mysql-operator/blob/f1cce241dc11eed4ee31aba2a3cfb638c4c03edb/lib/charms/mysql/v0/architecture.py#L60)), so it will never re-raise the exception. I can move the `raise` clause to an `else` block if that makes it easier to read.

Reply:
> I would check is_wrong_architecture before any imports

You mean even before the Standard library imports? Yeah, we can move it further up.

> I don't think we should raise since that will obscure the blocked status

The `raise` clause is there for cases where the import fails for any other reason (i.e. forget to change its name after a refactor, deploying in the wrong architecture but not actually executing the charm code, etc).

For the case that concern us, the `WrongArchitectureWarningCharm` class already exits with code `0` (see [code](https://github.com/canonical/mysql-operator/blob/f1cce241dc11eed4ee31aba2a3cfb638c4c03edb/lib/charms/mysql/v0/architecture.py#L60)), so it will never re-raise the exception. I can move the `raise` clause to an `else` block if that makes it easier.

Reply:
If you want to get rid of the `try-except` block (something I would also love to), we would need to tweak the `is_wrong_architecture` lib helper. Turns out unit tests do not set up the `CHARM_DIR` env. variable, and we assume that whenever we run that helper function it will always be available (see [code](https://github.com/canonical/mysql-operator/blob/f1cce241dc11eed4ee31aba2a3cfb638c4c03edb/lib/charms/mysql/v0/architecture.py#L65)).

Just noticed: doing what the PostgreSQL folks do in their repo would have worked (see [code](https://github.com/canonical/postgresql-k8s-operator/blob/70a96ce33eb0de054b9297e3bd558f6ece54c0b0/src/arch_utils.py#L29)) 😅 

```python
manifest_file_path = f"os.environ.get('CHARM_DIR')/manifest.yaml"
```

Reply:
Let me tweak the architecture lib to make it more resilient. No repo is using it so far, so I would say it is OK.